src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md

```diff
@@ -0,0 +1,108 @@
+
+## Deploying Jaeger Services
+
+We are going to use vstart for getting started with tracing.
+
+For this please ensure you have:
+
+1. Ceph compiled with `WITH_JAEGER=ON`.
+2. deployed jaeger backend services - to deploy the Jaeger Backend service using the vstart development cluster you just need to pass the `--jaeger` option.
+
+By following the above steps, you should be able to set up all the Jaeger services and access the JaegerUI at [http://localhost:16686](http://localhost:16686/).
+
+Read more: [https://www.jaegertracing.io/docs/1.20/getting-started/#all-in-one](https://www.jaegertracing.io/docs/1.20/getting-started/#all-in-one)
+
+![images/image10.png](images/image10.png)
+
+(JaegerUI at [http://localhost:16686](http://localhost:16686/) )
+
+If for some reason you are not able to access JaegerUI, retry manually running Jaeger services using:
+
+## Tracing Core Services
+
+Now that we have Jaeger Services up and running, let's take a look into Ceph’s core services tracing.
+
+Ceph currently can trace operations in the OSD I/O path and we'll be it demonstrating here:
+
+We'll perform a simple write operation, for which we are using rados bench write. You can use any other available write interface of your choice.
+
+On refreshing the JaegerUI ([http://localhost:16686](http://localhost:16686/)), you should be able to see osd-services in the services field, which specify the traced OSD processes in a lifetime of a request.
+
+![images/image4.gif](images/image4.gif)
+
+![images/image8.png](images/image8.png)
+
+Right below is the Operations field, which lists the spans collected, clicking on which shall give you traces containing it.
+
+![images/image5.png](images/image5.png)
+
+we see a list of the traces gathered by our recent write operation, and an x(time)-y(duration) graph with traces as data points.
+
+![images/image3.gif](images/image3.gif)
+
+![images/image2.png](images/image2.png)
+
+when you click on any of the traces, it should expand into all the spans and metadata stored in them.
+
+![images/image6.gif](images/image6.gif)
+
+We can store Ceph's metadata in the form of tags and logs while creating a span, as seen below; logs field carrying Ceph's relevant metadata.
+
+![images/image9.png](images/image9.png)
+
+There you have your internal processes visualized!
+
+**Exporting Traces as JSON**
+
+These traces are portable! They can be converted into JSON output, which can be rendered on other systems using the JaegerUI.
+
+![images/image7.png](images/image7.png)
+
+**Demo**
+
+<iframe
+  width="560"
+  height="315"
+  src="https://bluejeans.com/s/fcnR5lu_lBR"
+  title="Quick Demo of using JaegerUI for Ceph"
+  frameborder="0"
+  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+  allowfullscreen
+></iframe>
```

Comment:
For me, this Bluejeans recording has the screen split into two halves, with the meeting screen on the left and the actual demo on the right.  I would suggest something like asciinema instead -- much better suited for blog posts IMO.